libs/muddle/internal/muddle.hpp

```diff
@@ -104,20 +111,20 @@ class MuddleEndpoint;
  *                         Underlying Network Connections
  *
  */
-class Muddle
+class Muddle : public MuddleInterface, public std::enable_shared_from_this<Muddle>
```

Comment:
curious: when is muddle pointing to itself?

Reply:
When it registers itself with the registry. This is useful for collecting process wide stats on muddle instance.